packages/esbuild/esbuild.bzl

```diff
@@ -0,0 +1,220 @@
+
+    # disable the error limit and show all errors
+    args.add_joined(["--error-limit", "0"], join_with = "=")
+
+    if ctx.attr.minify:
+        args.add("--minify")
+    else:
+        # by default, esbuild will tree-shake 'pure' functions
+        # disable this unless also minifying
+        args.add_joined(["--tree-shaking", "ignore-annotations"], join_with = "=")
+
+    if ctx.attr.sources_content:
+        args.add("--sources-content=true")
+    else:
+        args.add("--sources-content=false")
+
+    if ctx.attr.splitting:
+        js_out = ctx.actions.declare_directory("%s" % ctx.attr.name)
+        outputs.append(js_out)
+
+        args.add("--splitting")
+        args.add_joined(["--format", "esm"], join_with = "=")
+        args.add_joined(["--outdir", js_out.path], join_with = "=")
+    else:
+        js_out = ctx.outputs.output
+        js_out_map = ctx.outputs.output_map
+        outputs.extend([js_out, js_out_map])
+
+        if ctx.attr.format:
+            args.add_joined(["--format", ctx.attr.format], join_with = "=")
+
+        args.add_joined(["--outfile", js_out.path], join_with = "=")
+
+    jsconfig_file = write_jsconfig_file(ctx, path_alias_mappings)
+    args.add_joined(["--tsconfig", jsconfig_file.path], join_with = "=")
+    inputs.append(jsconfig_file)
+
+    ctx.actions.run(
+        inputs = inputs,
+        outputs = outputs,
+        executable = ctx.executable.tool,
+        arguments = [args],
+        progress_message = "%s Javascript %s [esbuild]" % ("Bundling" if not ctx.attr.splitting else "Splitting", entry_point.short_path),
+        execution_requirements = {
+            "no-remote-exec": "1",
+        },
+    )
+
+    return [
+        DefaultInfo(files = depset(outputs + [jsconfig_file])),
+    ]
+
+esbuild_bundle = rule(
+    attrs = {
+        "define": attr.string_list(
+            default = [],
+            doc = """A list of global identifier replacements.
+Example:
+```python
+esbuild(
+    name = "bundle",
+    define = [
+        "process.env.NODE_ENV=\\"production\\""
+    ],
+)
+```
+            """,
+        ),
+        "deps": attr.label_list(
+            default = [],
+            aspects = [module_mappings_aspect, node_modules_aspect],
+            doc = "A list of direct dependencies that are required to build the bundle",
+        ),
+        "entry_point": attr.label(
+            mandatory = True,
+            allow_single_file = True,
+            doc = "The bundle's entry point (e.g. your main.js or app.js or index.js)",
+        ),
+        "external": attr.string_list(
+            default = [],
+            doc = "A list of module names that are treated as external and not included in the resulting bundle",
+        ),
+        "format": attr.string(
+            values = ["iife", "cjs", "esm", ""],
+            mandatory = False,
+            doc = """The output format of the bundle, defaults to iife when platform is browser
```

Comment:
could we add links to the esbuild documentation for each option that we pass through?